### ISSUE TITLE
[Bug] va-alert needs meaningful SR text to display status

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/vets-design-system-documentation#5046]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

The icons in the va-alert component do not expose meaningful text equivalents.

Currently, the pseudo-element used for the alert icon (e.g., warning, error, success) is purely decorative from a screen reader’s perspective. The screen reader announces only the visible heading text, such as:

> Your balance may be overdue

Sighted users, however, receive additional meaning through both icon and color treatment. Because the icon conveys status (e.g., warning, error, success), the absence of a text equivalent results in a loss of information for assistive technology users.

## What I expected to happen

Screen reader users should hear a message equivalent in meaning to the visual presentation.
For example:

> **Warning.** Your balance may be overdue.

Visually this looks like 

> ⚠️ Your balance may be overdue.

This provides parity of experience between sighted and non-sighted users.

## Steps to reproduce:

1. [Go to `<va-alert>` with visible status icon in storybook](https://design.va.gov/storybook/?path=/docs/components-va-alert-sign-in--docs)
2. Turn on a screen reader (e.g., NVDA or VoiceOver).
3. Move focus to the alert.
4. Listen to what’s announced.

**Actual:**
Only the visible text is announced.

**Expected:**
The alert’s type is also announced (e.g., “Warning”, “Success”, “Error”) via sr-text that is in the heading tag.

If the slim alert is used, the alert type would be attached to the `<p>` in the alert, and announced first.

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [X] This bug is blocking work currently in progress
- [ ] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

## Acceptance criteria
- [ ] In `va-alert`, screen readers announce the alert type before the message text.
- [ ] In `va-alert` with the slim prop, screen readers announce the alert type before the message text.
- [ ] SR-only text is visually hidden but accessible via programmatic name.
- [ ] No duplicated announcement (i.e., icon + text doesn’t cause double reading).
- [ ] Implementation verified across browsers and AT combinations (NVDA + Firefox, JAWS + Chrome, VoiceOver + Safari).
- [ ] Warning alerts announce "Warning Alert"
- [ ] Error alerts announce "Error Alert"
- [ ] Success alerts announce "Success Alert"
- [ ] Informational alerts announce "Information Alert"

## Related Staging review findings
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/121247
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/122540

## USWDS
I've also filed this as a bug with USWDS
- https://github.com/uswds/uswds/issues/6539

**Comment by HeatherRienks** [comment]
@mgleo -FYI

**Comment by jeana-adhoc** [comment]
Currently the Sr-text baked into the component is providing the context, given that you have wrapped the text in the alert with a `<p>`.  That is not changing for now, and you should be able to meet the concerns of the collab cycle.  

The guidance we are exploring, but is not published, nor final, *may* be to include "Warning" into the title of the alert, and if we go that route, we will remove the sr-text from the alert. If we do make that determination, there will be updates to the guidance, alignment with governance and ADE, announcements and timelines.

**Comment by jeana-adhoc** [comment]
@amyleadem and I have been meeting with USWDS to solve this problem together. We're currently reviewing a draft doc of guidance they have provided. ([Google doc](https://docs.google.com/document/d/1uEsF7KY3WHzDTOCZ4rcdIVSLJYVanmWGc10-_Wj3eqg/edit?usp=sharing) 🔒 )

After Amy and I have reviewed and taken into consideration our own alert usage to see how they are impacted, we'll meet with the content team next, as the updated direction is heavily dependent upon content.
